### PR TITLE
Update data models to used freezed

### PR DESCRIPTION
**Pull request by lockieRichter**
Updates the data models for book and book labels to used the freezed package (https://pub.dev/packages/freezed). This will make updating the models simpler and adds a ton of functionality for free.

Also updats the repositories to prepare for the book label UI work.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -52,3 +52,4 @@ UNTRANSLATED.txt
 
 # Generated code
 *.g.dart
+*.freezed.dart
```
Comment (shockbytes) [inline .gitignore:55]: How is the code generated? Do we need to run a command? Do we need to update our Github Actions too?

File: lib/src/data/book/entity/book.dart [hunk 2]
```diff
@@ -1,124 +1,52 @@
...
+    required String publishedDate,
+    required int position,
+    required String isbn,
+    required String? thumbnailAddress,
+    required int startDate,
+    required int endDate,
+
+    /// Actually `forLaterDate` and should not be confused with BookState.WISHLIST. This mishap
+    /// is due to the initial naming and cannot be changed without breaking prior backups. So, just
+    /// treat this as `forLaterDate` and everything is fine  /// Actually `forLaterDate` and should not be confused with BookState.WISHLIST. This mishap
```
Comment (shockbytes) [inline lib/src/data/book/entity/book.dart]: Seems like the comment got duplicated.

**Review by shockbytes — APPROVED** [review]
Only one question and one issue with a duplicated comment. Otherwise it looks good!

File: .gitignore (hunk 1 above)
Comment (lockieRichter) [reply .gitignore:55]: Same way as all our other generated code which is nice - by `build_runner`. `dart run build_runner build --delete-conflicting-outputs`

File: lib/src/data/book/entity/book.dart (hunk 2 above)
Comment (lockieRichter) [reply lib/src/data/book/entity/book.dart]: Willl fix up

**Pull request merged**